### PR TITLE
AEA-1615 Contract Test Tool

### PR DESCRIPTION
**Pull request by 5A11**
This PR adds a contract test tool and applies it onto the existing ERC1155 contract package tests for Ethereum.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by solarw — APPROVED** [review]
LGTM

File: aea/test_tools/test_contract.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            # load contract into sys modules
+            Contract.from_config(configuration)
+
+        cls._contract = contract_registry.make(str(configuration.public_id))
+
+        cls.contract_address = cls._deploy_ethereum_contract(
+            cls._contract, cls.ledger_api, cls.deployer_crypto, gas=5000000
+        )
+
+    @staticmethod
```
Comment (DavidMinarsch) [inline aea/test_tools/test_contract.py:99]: This should not be tied to Ethereum, it should accept an "identifier" and work for arbitrary ledgers.

File: aea/test_tools/test_contract.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains test case classes based on pytest for AEA contract testing."""
+from pathlib import Path
+from typing import Any, Dict, Optional, cast
+
+from aea_ledger_ethereum import EthereumCrypto
```
Comment (DavidMinarsch) [inline aea/test_tools/test_contract.py:23]: aea should not depend on plugins, that causes circularity

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

**Review by DavidMinarsch — COMMENTED** [review]
@5A11 I applied the required fixes to make it ledger independent. The next step would be to apply the same test class to `TestCosmWasmContract` in  `tests/test_packages/test_contracts/test_erc1155/test_contract.py`.

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**